docs/DeveloperGuide.md

```diff
@@ -1,34 +1,283 @@
 # Developer Guide
 
-## Design & Implementation
+## 1. Setting Up
+   * Prerequisites
+     * JDK 11 or above
+     * IntelliJ IDE
+   * Setting up the project in your computer
+     * Fork this repo, and clone the fork to your computer
+     * Open IntelliJ (if you are not in the welcome screen, click File > Close Project to close the existing project dialog first)
+     * Set up the correct JDK version for Gradle
+     * Click Configure > Project Defaults > Project Structure
+     * Click New…​ and find the directory of the JDK
+     * Click Import Project
+     * Locate the build.gradle file and select it. Click OK
+     * Click Open as Project
+     * Click OK to accept the default settings.
+   * Verifying the setup
+     * Run the jikan.jikan and try a few commands
+     * Run the tests and ensure they all pass.
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+## 2. Design
+![image_info](./pictures/ClassDiagram.png)
```

Comment:
As mentioned above, the UML diagram is pretty large. Aside from breaking down the UML into components,  perhaps you might want to look into attributes in certain classes that have already been shown with an association?  For example, you have the Parser class associated with StorageCleaner, and at the same time StorageCleaner is an attribute listed in the Parser Class. Perhaps you can consider omitting either the association or the attribute as mentioned in the 2113 website https://nus-cs2113-ay1920s2.github.io/website/se-book-adapted/chapters/uml.html#class-diagrams.